### PR TITLE
[Merged by Bors] - Split up the builder module into submodules

### PR DESCRIPTION
**Pull request by maltesander**
## Description

The current structure:
```
├── builder
│   ├── configmap.rs
│   ├── event.rs
│   ├── mod.rs
│   ├── pod
│   │   ├── container.rs
│   │   ├── mod.rs
│   │   ├── security.rs
│   │   ├── volume.rs
│   └── meta.rs

```

Please leave any ideas / suggestions on what else we might need or how the general structure should look like!

For me maybe the volumes can be summarized.

closes https://github.com/stackabletech/operator-rs/issues/317

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Changed
+
+- BREAKING: Split up the builder module into submodules ([#348]).
```
Comment (teozkr) [inline CHANGELOG.md:9]: Regarding breakage.. do we want to reexport the types in the `builder` module?

File: src/builder/pod/volume_source.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
+use k8s_openapi::api::core::v1::CSIVolumeSource;
+use std::collections::BTreeMap;
+
+#[derive(Clone)]
+pub struct SecretOperatorVolumeSourceBuilder {
```
Comment (teozkr) [inline src/builder/pod/volume_source.rs:5]: Looks like the rest of the `VolumeSource` stuff is in `pod::volume`, any reason not to put this there too?

File: src/builder/resource.rs [hunk 3]
```diff
@@ -0,0 +1,349 @@
+use crate::error::{Error, OperatorResult};
```
Comment (teozkr) [inline src/builder/resource.rs:1]: Nit: how about `meta` for this module?

**Review by teozkr — COMMENTED** [review]
Haven't gone through this with a super fine-toothed comb (since it's "just" a cut/paste job), but overall this LGTM.

File: src/builder/pod/volume_source.rs (hunk 2 above)
Comment (maltesander) [reply src/builder/pod/volume_source.rs:5]: No i think we can merge volume_mount in there as well.

File: src/builder/pod/volume_source.rs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+#[derive(Clone)]
+enum SecretOperatorVolumeScope {
+    Node,
+    Pod,
+    Service { name: String },
+}
+
+#[cfg(test)]
+mod tests {}
```
[the remark this answers is not in the document]
Comment (maltesander) [reply src/builder/pod/volume_source.rs:74]: Yeah.

File: src/builder/mod.rs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+#[deprecated(
+    since = "0.15.0",
+    note = "This is for compatibility. Please use `builder::pod::volume::*` in the future"
+)]
+pub use pod::volume::*;
+
+#[deprecated(
+    since = "0.15.0",
+    note = "This is for compatibility. Please use `builder::pod::*` in the future"
```
[the remark this answers is not in the document]
Comment (maltesander) [reply src/builder/mod.rs]: done

**Comment by teozkr** [comment]
Oh, too late :P

**Pull request closed without merging**